clement-moulin-frier/vivarium
# Load scenes from clients

## Description
Added a system to load scenes with controllers, using grpc.

## Related Issue (if applicable)
closes #63 

## How to Test
Launch the server
```
python3 scripts/run_server.py
```
Launch the Panel interface
```
panel serve scripts/run_interface.py --autoreload
```

instanciate a `NotebookController`
```py
from vivarium.controllers.notebook_controller import NotebookController
controller = NotebookController()
```
call the load_scene func
```py
controller.load_scene("session_1")
```
Now the scene should have changed and `controller.agents` should only show 1 agent.

**IMPORTANT:** Currently does not work with the web interface, you have to run it again to see the changes
